schemaregistry/config.go

```diff
@@ -0,0 +1,67 @@
+package schemaregistry
+
+/**
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// ConfigValue supports the following types:
+//  bool, int, string, any type with the standard String() interface
+type ConfigValue interface{}
+
+// ConfigMap is a map containing ConfigValues by key
+type ConfigMap map[string]ConfigValue
```

Comment:
A couple of things I don't like about the way we do config with the go client:

1. it's not strongly typed. this adds a lot of value IMO (intellisense documenting + preventing errors at IDE time), can be autogenerated in the difficult (librdkafka config) case, and is something everyone using the client touches (and it's first impressions).
2. we pass in a reference to a map, but the values are allocated on the heap, so this is pointless.

With the disclaimer that I don't know golang, I'm thinking we might want something like:

```
type MyConfig struct {
        // In the client case, autogenerated from https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
	BootstrapServers string
	MessageMaxBytes  int
	MaxInFlight      int
}

func NewMyConfig(config map[string]string) MyConfig {
	var r MyConfig
	for key, val := range config {
		r.Set(key, val)
	}
	return r
}

func (c *MyConfig) Set(key string, val string) error {
	// In the client case, autogenerated from https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
	switch key {
	case "bootstrap.servers":
		c.BootstrapServers = val
	case "message.max.bytes":
		c.MessageMaxBytes = strconv.Atoi(val)
	case "max.in.flight":
		c.MaxInFlight = strconv.Atoi(val)
	default:
		return errors.New("unknown config")
	}
	return nil
}

func (c *MyConfig) Get(key string) {
	// blah
}
```

where there is a different config type for the each client type, schema registry, and each serde type.

finally, i think we don't need to pass a pointer to this (even though it's not clear to me in go if it's passed on the stack or heap - it's not critical path so doesn't matter, i think a cleaner api is better).

now is the time to figure out the story for config going forward (and game plan for updating the client types, or not)